### PR TITLE
fix higher level API to plotting #1460

### PR DESCRIPTION
**Pull request by kpaharelau**
fix higher level API to plotting #1460

More simple way to create simlpe TimePlot was added.

Now you can try to use SimpleTimePlot (plot = new SimpleTimePlot(table, ["column1", "column2", “column3”, “column4”], [other args to control whatever axes, labels, colors])).
Use 1460.bkr (https://www.dropbox.com/s/lbb2r7qfknf8l0n/1460.bkr?dl=0) to try.

Notes

default the names for the lines and points to the same as the column
default the xLabel to "Time"
default usetoolTip and showLegend to true
it can select colors automatically using a generated nice colors

parameres :

timeColumn: determines what column will be used for a time axis. "time"  by default

colors:[X, Y, Z...]
each X could be [r,g,b] or just a string, like you can do with css
including names:
white
WHITE
lightGray
LIGHT_GRAY
gray  
GRAY
darkGray
DARK_GRAY
black 
BLACK
red 
RED
pink
PINK
orange
ORANGE
yellow
YELLOW
green 
GREEN
magenta
MAGENTA
cyan
CYAN
blue
BLUE

displayLineNames: display names for Lines
displayPointNames: display names for Points

displayLines: Lines will be added on the plot if true
displayPoints:  Points will be added on the plot if true

### CODE REVIEW

**Comment by scottdraves** [comment]
This looks great.  I am seeing a problem with the legend check boxes though.  Here's a really simple use of the API: http://sharing.beakernotebook.com/gist/anonymous/b02f32043b02cbaf003a
I then clicked both boxes for the "low" line, but it did not vanish.

**Comment by scottdraves** [comment]
Use less saturated colors by default (75%) and put them in this order:
blue red green purple cyan yellow 
also less bright, 85%.

first color is: 54 54 217

**Comment by kpaharelau** [comment]
This looks great. I am seeing a problem with the legend check boxes though. Here's a really simple use of the API: http://sharing.beakernotebook.com/gist/anonymous/b02f32043b02cbaf003a
I then clicked both boxes for the "low" line, but it did not vanish. -FIXED

Use less saturated colors by default (75%) and put them in this order:
blue red green purple cyan yellow
also less bright, 85%.

first color is: 54 54 217

I used collection nice colors for first 6 elements
//saturation 75%
//brightness 85%
private static final Color[] NICE_COLORS = {
    new Color(54, 54, 217),   //blue hue = 240
    new Color(216, 54, 54),   //red hue = 0
    new Color(54, 216, 170),  //green hue = 163
    new Color(192, 54, 216),  //purple hue = 291
    new Color(54, 126, 186),  //cyan hue = 169
    new Color(216, 178, 54),  //yellow hue = 46
}

next  elements gets generated color (random hue parameter, saturation 75% and brightness 85%).

But I have doubt that this is exactly what you need. What is your opinion? Probably better to use an algorithm from the XYChartSerializer?

 private static class ColorPalette {
    static final Color[] colors = new Color[]{
      new Color(192, 80, 77),
      new Color(79, 129, 189),
      new Color(155, 187, 89),
      new Color(247, 150, 70),
      new Color(128, 100, 162),
      new Color(75, 172, 198)
    };

```
 static Color getColor(int i) {
  return colors[i % colors.length];
}
```

  }

...
 int i = 0;
    for (XYGraphics g : xychart.getGraphics()) {
      if (g.getColor() == null) {
        g.setColori(ColorPalette.getColor(i++));
      }
    }
...

**Comment by scottdraves** [comment]
Thanks that's much better.  But green hue is off, it should be 120, not 163.
Don't worry about more than 6 colors this is fine for a start.
Continuing my review...

**Comment by kpaharelau** [comment]
But green hue is off, it should be 120, not 163. - fix

**Comment by scottdraves** [comment]
follow our style guide (indentation 2)

**Comment by kpaharelau** [comment]
follow our style guide (indentation 2) - fixed

File: plugin/jvm/src/main/java/com/twosigma/beaker/chart/xychart/SimpleTimePlot.java [hunk 1]
```diff
@@ -0,0 +1,385 @@
...
+        String colorAsStr = (String) color;
+        if (colorAsStr.indexOf("#") == 0) {
+          return Color.decode(colorAsStr);
+        } else {
+          Field field = Class.forName("com.twosigma.beaker.chart.Color").getField(colorAsStr);
+          return (Color) field.get(null);
+        }
+      }
+    } catch (Throwable ignored) {
+      //expected
```
Comment (scottdraves) [inline plugin/jvm/src/main/java/com/twosigma/beaker/chart/xychart/SimpleTimePlot.java]: what error is expected?  probably better to avoid any error.

**Comment by scottdraves** [comment]
make the default for displayPoints to false.
replace displayLineNames and displayPointNames with displayNames.

**Comment by kpaharelau** [comment]
User can make mistakes. For example 
 colors: ["#ffff00AA", "green", [64, 64], "rad"]

In this case color will be taken from nice colors or generated

**Comment by scottdraves** [comment]
If the user gives an illegal color then you should throw an error and say what is wrong.

**Comment by kpaharelau** [comment]
Ireplace displayLineNames and displayPointNames with displayNames. - Use same for lines and points?

If the user gives an illegal color then you should throw an error and say what is wrong. - Ok

**Comment by scottdraves** [comment]
> Use same for lines and points?

yes

**Comment by kpaharelau** [comment]
Color select algorithm has been improved according Scott remarks

make the default for displayPoints to false.
replace displayLineNames and displayPointNames with displayNames.  - FIXED

**Comment by scottdraves** [comment]
:+1:

**Pull request merged**